### ISSUE TITLE
[n4 sfos] pin code not recognized at first

### ISSUE THREAD

**Issue by rinigus** [rinigus/android#3]
There are problems with SIM card unlocking (SFOS versions 2.0.1.y and 2.0.2.x on CM12.1). When typing PIN once, SIM card is not unlocked. Press Enter second time, wait a bit, and after few seconds (5+ sec), the message "Incorrect ..." is displayed, dialog box disappears, and the card is unlocked. In few seconds after that, cellular connection is established.

When logging using logcat (radio and the main log), I didn't see anything during that time that would be out of the ordinary.

**Comment by ballock** [comment]
Checked on cm12.1 YOG4P with sfos 2.0.1.11. The behaviour is as follows: device boots, requests SIM card PIN, goes through, but does not establish cellular connection, even after a minute or so, does not re-issue the PIN screen.

Going to Settings -> Cellular shows a screen saying that no SIM cards are active, and I can activate the SIM from the pulley menu. This shows the PIN input screen, but does not accept the PIN. I tapped "Cancel" after the second attempt, it showed some error message about only being able to dial 112 on the same white screen with PIN input and went away, the data connection got established.
